src/main/java/ch/njol/skript/effects/EffLoadServerIcon.java

```
@@ -0,0 +1,91 @@
+import ch.njol.util.Kleenean;
+import org.bukkit.Bukkit;
+import org.bukkit.event.Event;
+import org.bukkit.util.CachedServerIcon;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Load Server Icon")
+@Description({"Loads server icons from the given files. You can get the loaded icon using the <a href='expressions.html#ExprLastLoadedServerIcon'>last loaded server icon</a> expression.",
+		"Please note that the image must be 64x64 and the file path starts from the server folder."})
+@Examples({"on load:",
+		"	clear {server-icons::*}",
+		"	loop 5 times:",
+		"		load server icon from file \"icons/%loop-number%.png\"",
+		"		add the last loaded server icon to {server-icons::*}",
+		"",
+		"on server list ping:",
+		"	set the icon to a random server icon out of {server-icons::*}"})
+@Since("INSERT VERSION")
+public class EffLoadServerIcon extends AsyncEffect {
+
+	static {
+		Skript.registerEffect(EffLoadServerIcon.class, "(load|cache) [the] server icon (of|from) [image] [file] %string%");
+	}
+
+	@SuppressWarnings("null")
+	private Expression<String> path;
+
+	@Nullable
+	public static CachedServerIcon lastLoaded = null;
+
+	@SuppressWarnings({"unchecked", "null"})
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
+		path = (Expression<String>) exprs[0];
+		return true;
+	}
+
+    @Override
+    protected void execute(Event e) {
+		File f = new File(path.getSingle(e));
+		if (f.exists() && f.isFile()) {
+			try {
+				lastLoaded = Bukkit.loadServerIcon(f);
+			} catch (NullPointerException npe) {
+				Skript.warning("'" + path + "' is not an image!");
+			} catch (IllegalArgumentException iae) {
+				Skript.warning("The image to be set as the server icon must be 64x64 pixels! ('" + path + "')");
+			} catch (Exception ex) {
+				Skript.exception(ex);
+			}
+		} else {
+			Skript.warning("'" + path.getSingle(e) + "' is not a valid file path!");
```

shouldn't this fail silently like everything else in skript? Also, don't call a get method more than once -- it can cause unneeded lag if people are using intensive expressions